### PR TITLE
Add ability to pre-fill the search bar from URL param

### PR DESCRIPTION
**Pull request by icyflame**
This will enable the addition of MFQP as a custom search engine in Chrome. (with a URL structure something like: `https://qp.metakgp.org/?search=%s`

I have tested this change, and it works well.

### CODE REVIEW

**Review by athityakumar — CHANGES_REQUESTED** [review]
A minor change required. 

Irrespective of the query in the url param, a `/` gets added at last - which messes up the search results at times.

Search as URL param : 
image

Search as input to search bar :
image

Notice the 2nd result missing in case 1. Maybe just the `/` can be removed from the `search_query` variable in [line 107](https://github.com/icyflame/mfqp/blob/add-search-url-param/index.html#L107)?

**Comment by athityakumar** [comment]
@icyflame - I have sent a [PR](https://github.com/icyflame/mfqp/pull/1) for the same. Have a look! :smile:

**Comment by icyflame** [comment]
image

I added that search engine to Chrome 56. Then, I searched using the address bar. There was no `/` that was appended.

> Irrespective of the query in the url param, a / gets added at last - which messes up the search results at times.

Who adds the `/`? If the user manually adds it, that's something that they have chosen to. I don't see the browser adding anything by default even when I manually type out the URL.

**Comment by athityakumar** [comment]
It gets added by the browser itself (I think. Can't think of any other reason), after the user has typed his search param and pressed the Enter button. For example,

Search param :
image

Result URL shows a `/` at the end :
image

I've added it as a search engine to Google Chrome Version 56.0.2924.87 (64-bit).

**Comment by icyflame** [comment]
@athityakumar Can you post the search engine entry here? You might have put a `/` after the `%s`. If you add the search engine URL as `http://localhost:8000/?search=%s` the `/` won't be appended.

In any case, it will be a good idea to remove trailing forward slashes. So, you should change your PR on the other issue to ensure that the last character is a forward slash before slicing it away. :smile:

**Comment by athityakumar** [comment]
image

Sure, I'll do the required changes on that PR. :smile:

**Comment by icyflame** [comment]
@athityakumar Actually, I want to look at the URL that you added in the search engine. That has been wrapped out of the frame.

**Comment by athityakumar** [comment]
@icyflame - Um, isn't it the `http://localhost:8000/?search=%s`?

**Comment by icyflame** [comment]
Hmm, okay, I guess your browser decided to append the `/` to the query. Mine doesn't. Weirdly enough, I am on the **exact same build of Chrome**: Version 56.0.2924.87 (64-bit)

Okay, I will merge that PR and that should update the commit here.

**Comment by athityakumar** [comment]
Weird o weird. :confused: 

I've updated the other PR. :smile:

**Comment by icyflame** [comment]
@athityakumar Merged. I have tested with multiple trailing forward slashes, works fine.

**Review by athityakumar — APPROVED** [review]

**Comment by athityakumar** [comment]
@icyflame - No more issues from my side, I've approved the PR. Let's wait for reviews of @vivekiitkgp & @amrav before merging this in. :smile:

**Comment by icyflame** [comment]
@vivekiitkgp and @amrav Please review this PR. :slightly_smiling_face:

I will merge this PR in a few hours. Thanks!

**Review by raivivek — APPROVED** [review]

**Pull request merged**
